tests/tests_pytorch/core/test_metric_result_integration.py

```diff
@@ -666,19 +668,27 @@ def on_train_start(self):
 
 
 @pytest.mark.parametrize("distributed_env", [True, False])
-def test_logger_sync_dist(distributed_env):
-    # self.log('bar', 7, ..., sync_dist=False)
+@pytest.mark.parametrize("log_val", [torch.tensor(0.5), torchmetrics.Accuracy()])
+def test_logger_sync_dist(distributed_env, log_val):
+    pl.trainer.connectors.logger_connector.result.warning_cache.clear()
+
+    # self.log('bar', 0.5, ..., sync_dist=False)
     meta = _Metadata("foo", "bar")
     meta.sync = _Sync(_should=False)
-    result_metric = _ResultMetric(metadata=meta, is_tensor=True)
-    result_metric.update(torch.tensor(7.0), 10)
+    is_tensor = isinstance(log_val, torch.Tensor)
+
+    if not is_tensor:
+        log_val.update(torch.Tensor([0, 1]), torch.LongTensor([0, 0]))
```

Comment:
The class constructors are not recommended

```suggestion
        log_val.update(torch.tensor([0, 1]), torch.tensor([0, 0], dtype=torch.long))
```